core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -17,40 +17,92 @@
+import { AppConstants } from 'app.constants';
+import { ContributorAdminDashboardFilter } from './contributor-admin-dashboard-filter.model';
 import { UserService } from 'services/user.service';
 
+
+export interface LanguageChoice {
+  id: string;
+  language: string;
+}
+
+export interface TopicChoice {
+  id: string;
+  topic: string;
+}
+
+export interface selectedTopicChoices {
+  ids: string[];
+  topics: string[];
+}
+
 @Component({
   selector: 'contributor-admin-dashboard-page',
   styleUrls: ['./contributor-admin-dashboard-page.component.css'],
   templateUrl: './contributor-admin-dashboard-page.component.html',
   animations: [
-    trigger('detailExpand', [
-      state('collapsed', style({height: '0px', minHeight: '0'})),
-      state('expanded', style({height: '*'})),
-      transition(
-        'expanded <=> collapsed',
-        animate('225ms cubic-bezier(0.4, 0.0, 0.2, 1)')),
+    trigger('lastActivityDropdown', [
+      state('expanded', style({ transform: 'rotate(180deg)' })),
+      state('collapsed', style({ transform: 'rotate(0)' })),
+      transition('expanded => collapsed', animate('200ms ease-out')),
+      transition('collapsed => expanded', animate('200ms ease-in')),
+    ]),
+    trigger('languageDropdownTrigger', [
+      state('expanded', style({ transform: 'rotate(180deg)' })),
+      state('collapsed', style({ transform: 'rotate(0)' })),
+      transition('expanded => collapsed', animate('200ms ease-out')),
+      transition('collapsed => expanded', animate('200ms ease-in')),
     ]),
   ],
 })
 export class ContributorAdminDashboardPageComponent implements OnInit {
+  @ViewChild(
+    'languageDropdown', {'static': false}) languageDropdownRef!: ElementRef;
+
+  @ViewChild(
+    'activityDropdown', {'static': false}) activityDropdownRef!: ElementRef;
+
+  languageDropdownShown: boolean = false;
+  activityDropdownShown: boolean = false;
   activeTab!: string;
   TAB_NAME_TRANSLATION_SUBMITTER: string = 'Translation Submitter';
   TAB_NAME_TRANSLATION_REVIEWER: string = 'Translation Reviewer';
   TAB_NAME_QUESTION_SUBMITTER: string = 'Question Submitter';
   TAB_NAME_QUESTION_REVIEWER: string = 'Question Reviewer';
+  translationReviewersCountByLanguage!: translationReviewersCount;
   translationReviewersCount: number = 0;
   questionReviewersCount: number = 0;
-  CONTRIBUTION_TYPES!: string[];
+  CONTRIBUTION_TYPES: string[] = [];
   selectedContributionType!: string;
   isQuestionCoordinator!: boolean;
   isTranslationCoordinator!: boolean;
+  loadingMessage!: string;
+  selectedLastActivity!: number;
+  allTopicNames: string[] = [];
+
+  languageChoices: LanguageChoice[] = [];
+  lastActivity: number[] = [];
+  filter: ContributorAdminDashboardFilter = (
+    ContributorAdminDashboardFilter.createDefault());
+
+  selectedTopics: selectedTopicChoices = {
+    ids: [],
+    topics: []
+  };
+
+  selectedLanguage: LanguageChoice = {
+    language: '',
+    id: ''
+  };
+
+  topics: TopicChoice[] = [];
```

Comment:
```suggestion
  languageChoices: LanguageChoice[] = [];
  lastActivity: number[] = [];
  filter: ContributorAdminDashboardFilter = (
    ContributorAdminDashboardFilter.createDefault());
  selectedTopics: selectedTopicChoices = {
    ids: [],
    topics: []
  };
  selectedLanguage: LanguageChoice = {
    language: '',
    id: ''
  };
  topics: TopicChoice[] = [];
```
Nit: The new lines aren't necessary and may even be confusing since it implies some logical separation when there isn't really one.

Reply:
If we don;t add blank lines between `selectedTopics` and `selectedLanguage` there's a lint error `Expected blank line between class members.`.